contracts/IdentityOracle.sol

```diff
@@ -83,66 +80,27 @@ contract IdentityOracle is DAOUpgradeableContract, ChainlinkClient, Ownable {
         _onlyOracle();
         lastStartUpdProcInvoked = block.timestamp;
         Chainlink.Request memory req = buildChainlinkRequest(
-            stringToBytes32(JOBID_GET_STATE_HASH),
+            stringToBytes32(JOBID_GET_STATE_HASH_IPFSCID),
             address(this),
-            this.setTmpStateHash.selector
+            this.setFulfillStateHashIPFSCID.selector
         );
-        sendChainlinkRequestTo(ORACLE_ADDRESS, req, ORACLE_PAYMENT);
+        requestOracleData(req, ORACLE_PAYMENT);
     }
 
     // It's the second function to be called by the oracle to store the tmpStateHash value
-    function setTmpStateHash(bytes32 _requestId, bytes32 _statehash)
-        public
-        recordChainlinkFulfillment(_requestId)
-    // This function is called only oracle
-    //- only approved oracles can set the new merkle state plus link to ipfs data used to create state
-    {
-        _onlyOracle();
-        tmpStateHash = _statehash;
-        Chainlink.Request memory req = buildChainlinkRequest(
-            stringToBytes32(JOBID_GET_IPFS_CID1),
-            address(this),
-            this.setTmpStateDataIPFS1.selector
-        );
-        sendChainlinkRequestTo(ORACLE_ADDRESS, req, ORACLE_PAYMENT);
-    }
-
-    // It's the third function to be called by the oracle to store the first 32 bytes of IPFS CID
-    function setTmpStateDataIPFS1(bytes32 _requestId, bytes32 _ipfscid1)
+    function setFulfillStateHashIPFSCID(bytes32 _requestId, bytes memory _statehashipfscid)
```

Comment:
is the method  setState() now redundant?

Reply:
fixed by https://github.com/GoodDollar/ChainlinkIdentityOracle/pull/5/commits/f05ce99726a869f1dfaf944456647138eaf25fa3